src/Beckett/Storage/Postgres/ServiceCollectionExtensions.cs

```diff
@@ -13,12 +14,22 @@ public static void AddPostgresSupport(this IServiceCollection services, BeckettO
             return;
         }
 
-        services.AddSingleton<IPostgresDatabase, PostgresDatabase>();
+        services.AddSingleton<IPostgresDatabase>(provider =>
+        {
+            var dataSource = options.Postgres.DataSource ?? provider.GetService<NpgsqlDataSource>();
+            if (dataSource is null)
+            {
+                throw new InvalidOperationException(
+                    "Registered NpgsqlDataSource not found - please register one using AddNpgsqlDataSource from the Npgsql.DependencyInjection package, provide a configured instance via UseDataSource, or call UseConnectionString"
+                );
+            }
+            return new PostgresDatabase(dataSource);
+        });
 
         services.AddSingleton<IEventStorage, PostgresEventStorage>();
 
         services.AddSingleton<IPostgresNotificationListener, PostgresNotificationListener>();
 
-        services.AddSingleton<ISubscriptionStorage, PostgresSubscriptionStorage>();
+        services.AddSingleton<ISubscriptionStorage, PostgresSubscriptionStorage>(); ;
```

Comment:
so many semicolons...
```suggestion
        services.AddSingleton<ISubscriptionStorage, PostgresSubscriptionStorage>();
```